Adopt more smart pointers in the Document class

cdumez opened the pull request:
#### fa36ebb5c3c7bffd7215478e377baeb89ebc86da
<pre>
Adopt more smart pointers in the Document class
<a href="https://bugs.webkit.org/show_bug.cgi?id=263342">https://bugs.webkit.org/show_bug.cgi?id=263342</a>

Reviewed by Ryosuke Niwa.

* Source/WebCore/dom/Document.cpp:
(WebCore::CallbackForContainIntrinsicSize):
(WebCore::printNavigationErrorMessage):
(WebCore::Document::configureSharedLogger):
(WebCore::Document::addToDocumentsMap):
(WebCore::Document::allDocumentsMap):
(WebCore::createCachedResourceLoader):
(WebCore::m_frameIdentifier):
(WebCore::Document::~Document):
(WebCore::Document::commonTeardown):
(WebCore::Document::setCompatibilityMode):
(WebCore::Document::childrenChanged):
(WebCore::createUpgradeCandidateElement):
(WebCore::createHTMLElementWithNameValidation):
(WebCore::Document::createCSSStyleDeclaration):
(WebCore::createFallbackHTMLElement):
(WebCore::Document::createElement):
(WebCore::Document::documentEventTimingFromNavigationTiming):
(WebCore::Document::setVisualUpdatesAllowed):
(WebCore::Document::setCharset):
(WebCore::Document::setDocumentElementLanguage):
(WebCore::Document::protectedParser const):
(WebCore::Document::setContent):
(WebCore::Document::contentType const):
(WebCore::Document::caretPositionFromPoint):
(WebCore::Document::scrollingElement):
(WebCore::canonicalizedTitle):
(WebCore::Document::updateTitle):
(WebCore::Document::updateTitleFromTitleElement):
(WebCore::Document::setTitle):
(WebCore::Document::protectedTitleElement const):
(WebCore::Document::updateTitleElement):
(WebCore::Document::resolveStyle):
(WebCore::Document::updateLayout):
(WebCore::Document::updateLayoutIfDimensionsOutOfDate):
(WebCore::Document::isResolvingContainerQueriesForSelfOrAncestor const):
(WebCore::Document::createRenderTree):
(WebCore::Document::didBecomeCurrentDocumentInFrame):
(WebCore::Document::willDetachPage):
(WebCore::Document::attachToCachedFrame):
(WebCore::Document::protectedDocumentElement const):
(WebCore::Document::destroyRenderTree):
(WebCore::Document::protectedUndoManager const):
(WebCore::Document::willBeRemovedFromFrame):
(WebCore::Document::protectedReportingScope const):
(WebCore::Document::removeAllEventListeners):
(WebCore::Document::axObjectCache const):
(WebCore::Document::setVisuallyOrdered):
(WebCore::Document::createParser):
(WebCore::Document::appHighlightRegistry):
(WebCore::Document::updateHighlightPositions):
(WebCore::Document::openForBindings):
(WebCore::Document::open):
(WebCore::Document::isFullyActive const):
(WebCore::Document::detachParser):
(WebCore::Document::cancelParsing):
(WebCore::Document::implicitOpen):
(WebCore::Document::fontLoadRequest):
(WebCore::Document::beginLoadingFontSoon):
(WebCore::Document::setBodyOrFrameset):
(WebCore::Document::explicitClose):
(WebCore::Document::implicitClose):
(WebCore::Document::setParsing):
(WebCore::Document::enqueuePaintTimingEntryIfNeeded):
(WebCore::Document::write):
(WebCore::Document::minimumDOMTimerInterval const):
(WebCore::Document::domTimerAlignmentInterval const):
(WebCore::Document::urlForBindings const):
(WebCore::Document::fallbackBaseURL const):
(WebCore::Document::processBaseElement):
(WebCore::Document::userAgent const):
(WebCore::Document::disableEval):
(WebCore::Document::disableWebAssembly):
(WebCore::Document::idbConnectionProxy):
(WebCore::Document::createRTCDataChannelRemoteHandlerConnection):
(WebCore::Document::canNavigateInternal):
(WebCore::Document::isNavigationBlockedByThirdPartyIFrameRedirectBlocking):
(WebCore::Document::dispatchDisabledAdaptationsDidChangeForMainFrame):
(WebCore::Document::viewportArguments const):
(WebCore::Document::updateViewportArguments):
(WebCore::Document::themeColorChanged):
(WebCore::Document::processColorScheme):
(WebCore::Document::metaElementColorSchemeChanged):
(WebCore::Document::processWebAppOrientations):
(WebCore::Document::prepareMouseEvent):
(WebCore::Document::canAcceptChild const):
(WebCore::Document::cloneNodeInternal):
(WebCore::Document::cloneDataFromDocument):
(WebCore::Document::protectedDecoder const):
(WebCore::Document::runScrollSteps):
(WebCore::Document::noteUserInteractionWithMediaElement):
(WebCore::Document::updateIsPlayingMedia):
(WebCore::Document::flushAutofocusCandidates):
(WebCore::Document::hoveredElementDidDetach):
(WebCore::Document::updateEventRegions):
(WebCore::Document::scheduleDeferredAXObjectCacheUpdate):
(WebCore::Document::flushDeferredAXObjectCacheUpdate):
(WebCore::Document::updateAccessibilityObjectRegions):
(WebCore::Document::invalidateEventRegionsForFrame):
(WebCore::Document::invalidateEventListenerRegions):
(WebCore::Document::invalidateRenderingDependentRegions):
(WebCore::Document::protectedFocusedElement const):
(WebCore::Document::setFocusedElement):
(WebCore::Document::focusNavigationStartingNode const):
(WebCore::Document::updateRangesAfterChildrenChanged):
(WebCore::Document::nodeChildrenWillBeRemoved):
(WebCore::Document::nodeWillBeRemoved):
(WebCore::Document::parentlessNodeMovedToNewDocument):
(WebCore::Document::textInserted):
(WebCore::Document::textRemoved):
(WebCore::Document::textNodesMerged):
(WebCore::Document::textNodeSplit):
(WebCore::Document::takeDOMWindowFrom):
(WebCore::Document::protectedContextDocument const):
(WebCore::Document::setWindowAttributeEventListener):
(WebCore::Document::dispatchWindowEvent):
(WebCore::Document::dispatchWindowLoadEvent):
(WebCore::Document::queueTaskToDispatchEventOnWindow):
(WebCore::Document::enqueueOverflowEvent):
(WebCore::Document::cookie):
(WebCore::Document::setCookie):
(WebCore::Document::referrerForBindings):
(WebCore::Document::shouldMaskURLForBindingsInternal const):
(WebCore::Document::setBackForwardCacheState):
(WebCore::Document::suspend):
(WebCore::Document::resume):
(WebCore::Document::requiresUserGestureForAudioPlayback const):
(WebCore::Document::requiresUserGestureForVideoPlayback const):
(WebCore::Document::mediaDataLoadsAutomatically const):
(WebCore::Document::privateBrowsingStateDidChange):
(WebCore::Document::registerForCaptionPreferencesChangedCallbacks):
(WebCore::Document::captionPreferencesChanged):
(WebCore::Document::updateTextTrackRepresentationImageIfNeeded):
(WebCore::Document::applyPendingXSLTransformsTimerFired):
(WebCore::Document::finishedParsing):
(WebCore::Document::updateURLForPushOrReplaceState):
(WebCore::Document::initDNSPrefetch):
(WebCore::Document::getParserLocation const):
(WebCore::Document::addConsoleMessage):
(WebCore::Document::addMessage):
(WebCore::Document::postTask):
(WebCore::Document::suspendScheduledTasks):
(WebCore::Document::resumeScheduledTasks):
(WebCore::Document::suspendScriptedAnimationControllerCallbacks):
(WebCore::Document::resumeScriptedAnimationControllerCallbacks):
(WebCore::Document::serviceRequestAnimationFrameCallbacks):
(WebCore::Document::serviceRequestVideoFrameCallbacks):
(WebCore::Document::exitPointerLock):
(WebCore::Document::loadEventDelayTimerFired):
(WebCore::Document::monotonicTimestamp const):
(WebCore::Document::clearScriptedAnimationController):
(WebCore::Document::wheelEventHandlersChanged):
(WebCore::Document::didRemoveEventTargetNode):
(WebCore::Document::didAddOrRemoveMouseEventHandler):
(WebCore::Document::updateLastHandledUserGestureTimestamp):
(WebCore::Document::processingUserGestureForMedia const):
(WebCore::Document::allowsContentJavaScript const):
(WebCore::Document::convertAbsoluteToClientQuads):
(WebCore::Document::convertAbsoluteToClientRects):
(WebCore::Document::convertAbsoluteToClientRect):
(WebCore::Document::decrementActiveParserCount):
(WebCore::DocumentParserYieldToken::~DocumentParserYieldToken):
(WebCore::findNearestCommonComposedAncestor):
(WebCore::Document::updateHoverActiveState):
(WebCore::Document::ensureTemplateDocument):
(WebCore::Document::documentFragmentForInnerOuterHTML):
(WebCore::Document::deviceScaleFactor const):
(WebCore::Document::useSystemAppearance const):
(WebCore::Document::useDarkAppearance const):
(WebCore::Document::useElevatedUserInterfaceLevel const):
(WebCore::Document::compositeOperatorForBackgroundColor const):
(WebCore::Document::didAssociateFormControl):
(WebCore::Document::didAssociateFormControlsTimerFired):
(WebCore::Document::didLoadResourceSynchronously):
(WebCore::Document::ensurePlugInsInjectedScript):
(WebCore::Document::wrapCryptoKey):
(WebCore::Document::unwrapCryptoKey):
(WebCore::Document::hasFocus const):
(WebCore::Document::addPlaybackTargetPickerClient):
(WebCore::Document::removePlaybackTargetPickerClient):
(WebCore::Document::showPlaybackTargetPicker):
(WebCore::Document::playbackTargetPickerClientStateDidChange):
(WebCore::Document::shouldOpenExternalURLsPolicyToPropagate const):
(WebCore::Document::shouldEnforceHTTP09Sandbox const):
(WebCore::Document::shouldEnforceQuickLookSandbox const):
(WebCore::Document::applyQuickLookSandbox):
(WebCore::Document::shouldEnforceContentDispositionAttachmentSandbox const):
(WebCore::Document::scheduleRenderingUpdate):
(WebCore::Document::updateIntersectionObservations):
(WebCore::Document::gatherResizeObservations):
(WebCore::Document::deliverResizeObservations):
(WebCore::Document::setHasSkippedResizeObservations):
(WebCore::Document::setDir):
(WebCore::Document::fgColor const):
(WebCore::Document::alinkColor const):
(WebCore::Document::linkColorForBindings const):
(WebCore::Document::vlinkColor const):
(WebCore::Document::logger):
(WebCore::Document::updateAnimationsAndSendEvents):
(WebCore::Document::matchingAnimations):
(WebCore::Document::hideAllPopoversUntil):
(WebCore::Document::handlePopoverLightDismiss):
(WebCore::Document::didLogMessage):
(WebCore::Document::setServiceWorkerConnection):
(WebCore::Document::updateServiceWorkerClientData):
(WebCore::Document::navigateFromServiceWorker):
(WebCore::Document::frameWasDisconnectedFromOwner):
(WebCore::Document::hitTest):
(WebCore::Document::setAsRunningUserScripts):
(WebCore::Document::setPictureInPictureElement):
(WebCore::Document::notificationClient):
(WebCore::Document::graphicsClient):
(WebCore::Document::sessionID const):
(WebCore::Document::observeForContainIntrinsicSize):
(WebCore::Document::unobserveForContainIntrinsicSize):
(WebCore::Document::resetObservationSizeForContainIntrinsicSize):
(WebCore::Document::noiseInjectionPolicy const):
(WebCore::Document::noiseInjectionHashSalt const):
(WebCore::Document::mediaKeysStorageDirectory):
(WebCore::Document::protectedWindow const):
(WebCore::Document::protectedCachedResourceLoader const):
(WebCore::Document::protectedFontSelector const):
* Source/WebCore/dom/Document.h:
(WebCore::Document::setContextDocument):
(WebCore::Document::protectedParentDocument const):
(WebCore::Document::protectedTopDocument const):
* Source/WebCore/dom/Node.cpp:
(WebCore::Node::~Node):
* Source/WebCore/fileapi/BlobURL.cpp:
(WebCore::blobOwner):
* Source/WebCore/layout/layouttree/LayoutTreeBuilder.cpp:
(WebCore::Layout::printLayoutTreeForLiveDocuments):
* Source/WebCore/page/FrameDestructionObserver.h:
* Source/WebCore/page/FrameDestructionObserverInlines.h:
(WebCore::FrameDestructionObserver::protectedFrame const):
* Source/WebCore/page/LocalFrame.cpp:
(WebCore::Document::protectedView const):
* Source/WebCore/page/LocalFrameView.cpp:
(WebCore::LocalFrameView::protectedFrame const):
* Source/WebCore/page/LocalFrameView.h:
* Source/WebCore/page/MemoryRelease.cpp:
(WebCore::releaseNoncriticalMemory):
(WebCore::releaseCriticalMemory):
* Source/WebCore/page/mac/PageMac.mm:
(WebCore::Page::platformInitialize):
* Source/WebCore/rendering/HitTestResult.cpp:
(WebCore::HitTestResult::setScrollbar):
* Source/WebCore/rendering/HitTestResult.h:
* Source/WebCore/rendering/RenderObject.cpp:
(WebCore::printPaintOrderTreeForLiveDocuments):
(WebCore::printRenderTreeForLiveDocuments):
(WebCore::printLayerTreeForLiveDocuments):
(WebCore::printGraphicsLayerTreeForLiveDocuments):
* Source/WebCore/rendering/RenderView.h:
* Source/WebCore/testing/Internals.cpp:
(WebCore::Internals::allSVGSVGElements const):
(WebCore::Internals::numberOfLiveNodes const):
* Source/WebCore/workers/service/SWClientConnection.cpp:
(WebCore::SWClientConnection::postTaskForJob):
(WebCore::postMessageToContainer):
(WebCore::SWClientConnection::postMessageToServiceWorkerClient):
(WebCore::SWClientConnection::updateRegistrationState):
(WebCore::SWClientConnection::updateWorkerState):
(WebCore::SWClientConnection::fireUpdateFoundEvent):
(WebCore::SWClientConnection::setRegistrationLastUpdateTime):
(WebCore::SWClientConnection::setRegistrationUpdateViaCache):
(WebCore::SWClientConnection::updateBackgroundFetchRegistration):
(WebCore::updateController):
(WebCore::SWClientConnection::notifyClientsOfControllerChange):
(WebCore::SWClientConnection::clearPendingJobs):
(WebCore::SWClientConnection::registerServiceWorkerClients):
* Source/WebKit/WebProcess/InjectedBundle/InjectedBundle.cpp:
(WebKit::InjectedBundle::liveDocumentURLs):

Canonical link: <a href="https://commits.webkit.org/269557@main">https://commits.webkit.org/269557@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/62aa0297e1057c5ec0d5ff491548d99ed472c815

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/22878 "Passed style check") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/881 "Build is being retried. Recent messages:OS: Sonoma (14.0), Xcode: 15.0") | [  ~~🛠 mac~~](https://ews-build.webkit.org/#/builders/14/builds/23974 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🛠 wpe~~](https://ews-build.webkit.org/#/builders/5/builds/24787 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/21178 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/23137 "Passed tests") | [  ~~🛠 ios-sim~~](https://ews-build.webkit.org/#/builders/49/builds/1989 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🛠 mac-AS-debug~~](https://ews-build.webkit.org/#/builders/16/builds/23413 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/5/builds/24787 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/23118 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/49/builds/1989 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/19849 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🛠 gtk~~](https://ews-build.webkit.org/#/builders/2/builds/25645 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/49/builds/1989 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/14/builds/23974 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/26933 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/49/builds/1989 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/14/builds/23974 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/24784 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/44/builds/415 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/16/builds/23413 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/46/builds/337 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/14/builds/23974 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/43/builds/832 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/45/builds/588 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

rniwa commented on Source/WebCore/dom/Document.cpp:
```diff
@@ -3168,12 +3189,12 @@ ExceptionOr<void> Document::open(Document* entryDocument)
-        if (m_frame->loader().policyChecker().delegateIsDecidingNavigationPolicy())
-            m_frame->loader().policyChecker().stopCheck();
+        bool isNavigating = frame->loader().policyChecker().delegateIsDecidingNavigationPolicy() || frame->loader().state() == FrameState::Provisional || frame->navigationScheduler().hasQueuedNavigation();
+        if (frame->loader().policyChecker().delegateIsDecidingNavigationPolicy())
```
Maybe we should store policyChecker in CheckedRef?

rniwa commented on Source/WebCore/dom/Document.cpp:
```diff
@@ -3253,23 +3272,23 @@ void Document::implicitOpen()
 std::unique_ptr<FontLoadRequest> Document::fontLoadRequest(const String& url, bool isSVG, bool isInitiatingElementInUserAgentShadowTree, LoadedFromOpaqueSource loadedFromOpaqueSource)
 {
-    auto* cachedFont = m_fontLoader->cachedFont(completeURL(url), isSVG, isInitiatingElementInUserAgentShadowTree, loadedFromOpaqueSource);
+    CachedResourceHandle<CachedFont> cachedFont = m_fontLoader->cachedFont(completeURL(url), isSVG, isInitiatingElementInUserAgentShadowTree, loadedFromOpaqueSource);
```
We can't just say CachedResourceHandle?

cdumez replied:
I do not remember if there was a reason for it or not. I'll try to drop it.

cdumez replied:
I missed this. I'll re-upload.

rniwa commented on Source/WebCore/dom/Document.cpp:
```diff
@@ -3767,70 +3786,73 @@ void Document::setBaseURLOverride(const URL& url)
 {
-    return frame() ? frame()->loader().userAgent(url) : String();
+    RefPtr frame = this->frame();
+    return frame ? frame->loader().userAgent(url) : String();
```
I think we should be storing loader() in a RefPtr/CheckedPtr since that's the object on which we call a non-trivial function: userAgent. loader() is a trivial function for all intended purposes.

cdumez replied:
FrameLoader is not refcounted. The FrameLoader is owned by the frame so ref'ing the frame (like I did) is what keeps the FrameLoader alive.

rniwa replied:
Can we use a CheckedRef for FrameLoader in that case?

rniwa commented on Source/WebCore/dom/Document.cpp:
```diff
@@ -3767,70 +3786,73 @@ void Document::setBaseURLOverride(const URL& url)
         return;
 
-    frame()->script().setEvalEnabled(false, errorMessage);
+    frame->script().setEvalEnabled(false, errorMessage);
```
Ditto, here too we probably want to store script in Ref or CheckedRef.

cdumez replied:
The script is owned by the Frame via UniqueRef<ScriptController> so ref'ing the frame like I did keeps it alive.

I guess we could also have ScriptController subclass CanMakeCheckedPtr and use a CheckedRef here. Do you think it is worth it?

rniwa commented on Source/WebCore/dom/Document.cpp:
```diff
@@ -3767,70 +3786,73 @@ void Document::setBaseURLOverride(const URL& url)
         return;
 
-    frame()->script().setWebAssemblyEnabled(false, errorMessage);
+    frame->script().setWebAssemblyEnabled(false, errorMessage);
```
Ditto.

cdumez committed:
Adopt more smart pointers in the Document class
https://bugs.webkit.org/show_bug.cgi?id=263342

Reviewed by Ryosuke Niwa.

* Source/WebCore/dom/Document.cpp:
(WebCore::CallbackForContainIntrinsicSize):
(WebCore::printNavigationErrorMessage):
(WebCore::Document::configureSharedLogger):
(WebCore::Document::addToDocumentsMap):
(WebCore::Document::allDocumentsMap):
(WebCore::createCachedResourceLoader):
(WebCore::m_frameIdentifier):
(WebCore::Document::~Document):
(WebCore::Document::commonTeardown):
(WebCore::Document::setCompatibilityMode):
(WebCore::Document::childrenChanged):
(WebCore::createUpgradeCandidateElement):
(WebCore::createHTMLElementWithNameValidation):
(WebCore::Document::createCSSStyleDeclaration):
(WebCore::createFallbackHTMLElement):
(WebCore::Document::createElement):
(WebCore::Document::documentEventTimingFromNavigationTiming):
(WebCore::Document::setVisualUpdatesAllowed):
(WebCore::Document::setCharset):
(WebCore::Document::setDocumentElementLanguage):
(WebCore::Document::protectedParser const):
(WebCore::Document::setContent):
(WebCore::Document::contentType const):
(WebCore::Document::caretPositionFromPoint):
(WebCore::Document::scrollingElement):
(WebCore::canonicalizedTitle):
(WebCore::Document::updateTitle):
(WebCore::Document::updateTitleFromTitleElement):
(WebCore::Document::setTitle):
(WebCore::Document::protectedTitleElement const):
(WebCore::Document::updateTitleElement):
(WebCore::Document::resolveStyle):
(WebCore::Document::updateLayout):
(WebCore::Document::updateLayoutIfDimensionsOutOfDate):
(WebCore::Document::isResolvingContainerQueriesForSelfOrAncestor const):
(WebCore::Document::createRenderTree):
(WebCore::Document::didBecomeCurrentDocumentInFrame):
(WebCore::Document::willDetachPage):
(WebCore::Document::attachToCachedFrame):
(WebCore::Document::protectedDocumentElement const):
(WebCore::Document::destroyRenderTree):
(WebCore::Document::protectedUndoManager const):
(WebCore::Document::willBeRemovedFromFrame):
(WebCore::Document::protectedReportingScope const):
(WebCore::Document::removeAllEventListeners):
(WebCore::Document::axObjectCache const):
(WebCore::Document::setVisuallyOrdered):
(WebCore::Document::createParser):
(WebCore::Document::appHighlightRegistry):
(WebCore::Document::updateHighlightPositions):
(WebCore::Document::openForBindings):
(WebCore::Document::open):
(WebCore::Document::isFullyActive const):
(WebCore::Document::detachParser):
(WebCore::Document::cancelParsing):
(WebCore::Document::implicitOpen):
(WebCore::Document::fontLoadRequest):
(WebCore::Document::beginLoadingFontSoon):
(WebCore::Document::setBodyOrFrameset):
(WebCore::Document::explicitClose):
(WebCore::Document::implicitClose):
(WebCore::Document::setParsing):
(WebCore::Document::enqueuePaintTimingEntryIfNeeded):
(WebCore::Document::write):
(WebCore::Document::minimumDOMTimerInterval const):
(WebCore::Document::domTimerAlignmentInterval const):
(WebCore::Document::urlForBindings const):
(WebCore::Document::fallbackBaseURL const):
(WebCore::Document::processBaseElement):
(WebCore::Document::userAgent const):
(WebCore::Document::disableEval):
(WebCore::Document::disableWebAssembly):
(WebCore::Document::idbConnectionProxy):
(WebCore::Document::createRTCDataChannelRemoteHandlerConnection):
(WebCore::Document::canNavigateInternal):
(WebCore::Document::isNavigationBlockedByThirdPartyIFrameRedirectBlocking):
(WebCore::Document::dispatchDisabledAdaptationsDidChangeForMainFrame):
(WebCore::Document::viewportArguments const):
(WebCore::Document::updateViewportArguments):
(WebCore::Document::themeColorChanged):
(WebCore::Document::processColorScheme):
(WebCore::Document::metaElementColorSchemeChanged):
(WebCore::Document::processWebAppOrientations):
(WebCore::Document::prepareMouseEvent):
(WebCore::Document::canAcceptChild const):
(WebCore::Document::cloneNodeInternal):
(WebCore::Document::cloneDataFromDocument):
(WebCore::Document::protectedDecoder const):
(WebCore::Document::runScrollSteps):
(WebCore::Document::noteUserInteractionWithMediaElement):
(WebCore::Document::updateIsPlayingMedia):
(WebCore::Document::flushAutofocusCandidates):
(WebCore::Document::hoveredElementDidDetach):
(WebCore::Document::updateEventRegions):
(WebCore::Document::scheduleDeferredAXObjectCacheUpdate):
(WebCore::Document::flushDeferredAXObjectCacheUpdate):
(WebCore::Document::updateAccessibilityObjectRegions):
(WebCore::Document::invalidateEventRegionsForFrame):
(WebCore::Document::invalidateEventListenerRegions):
(WebCore::Document::invalidateRenderingDependentRegions):
(WebCore::Document::protectedFocusedElement const):
(WebCore::Document::setFocusedElement):
(WebCore::Document::focusNavigationStartingNode const):
(WebCore::Document::updateRangesAfterChildrenChanged):
(WebCore::Document::nodeChildrenWillBeRemoved):
(WebCore::Document::nodeWillBeRemoved):
(WebCore::Document::parentlessNodeMovedToNewDocument):
(WebCore::Document::textInserted):
(WebCore::Document::textRemoved):
(WebCore::Document::textNodesMerged):
(WebCore::Document::textNodeSplit):
(WebCore::Document::takeDOMWindowFrom):
(WebCore::Document::protectedContextDocument const):
(WebCore::Document::setWindowAttributeEventListener):
(WebCore::Document::dispatchWindowEvent):
(WebCore::Document::dispatchWindowLoadEvent):
(WebCore::Document::queueTaskToDispatchEventOnWindow):
(WebCore::Document::enqueueOverflowEvent):
(WebCore::Document::cookie):
(WebCore::Document::setCookie):
(WebCore::Document::referrerForBindings):
(WebCore::Document::shouldMaskURLForBindingsInternal const):
(WebCore::Document::setBackForwardCacheState):
(WebCore::Document::suspend):
(WebCore::Document::resume):
(WebCore::Document::requiresUserGestureForAudioPlayback const):
(WebCore::Document::requiresUserGestureForVideoPlayback const):
(WebCore::Document::mediaDataLoadsAutomatically const):
(WebCore::Document::privateBrowsingStateDidChange):
(WebCore::Document::registerForCaptionPreferencesChangedCallbacks):
(WebCore::Document::captionPreferencesChanged):
(WebCore::Document::updateTextTrackRepresentationImageIfNeeded):
(WebCore::Document::applyPendingXSLTransformsTimerFired):
(WebCore::Document::finishedParsing):
(WebCore::Document::updateURLForPushOrReplaceState):
(WebCore::Document::initDNSPrefetch):
(WebCore::Document::getParserLocation const):
(WebCore::Document::addConsoleMessage):
(WebCore::Document::addMessage):
(WebCore::Document::postTask):
(WebCore::Document::suspendScheduledTasks):
(WebCore::Document::resumeScheduledTasks):
(WebCore::Document::suspendScriptedAnimationControllerCallbacks):
(WebCore::Document::resumeScriptedAnimationControllerCallbacks):
(WebCore::Document::serviceRequestAnimationFrameCallbacks):
(WebCore::Document::serviceRequestVideoFrameCallbacks):
(WebCore::Document::exitPointerLock):
(WebCore::Document::loadEventDelayTimerFired):
(WebCore::Document::monotonicTimestamp const):
(WebCore::Document::clearScriptedAnimationController):
(WebCore::Document::wheelEventHandlersChanged):
(WebCore::Document::didRemoveEventTargetNode):
(WebCore::Document::didAddOrRemoveMouseEventHandler):
(WebCore::Document::updateLastHandledUserGestureTimestamp):
(WebCore::Document::processingUserGestureForMedia const):
(WebCore::Document::allowsContentJavaScript const):
(WebCore::Document::convertAbsoluteToClientQuads):
(WebCore::Document::convertAbsoluteToClientRects):
(WebCore::Document::convertAbsoluteToClientRect):
(WebCore::Document::decrementActiveParserCount):
(WebCore::DocumentParserYieldToken::~DocumentParserYieldToken):
(WebCore::findNearestCommonComposedAncestor):
(WebCore::Document::updateHoverActiveState):
(WebCore::Document::ensureTemplateDocument):
(WebCore::Document::documentFragmentForInnerOuterHTML):
(WebCore::Document::deviceScaleFactor const):
(WebCore::Document::useSystemAppearance const):
(WebCore::Document::useDarkAppearance const):
(WebCore::Document::useElevatedUserInterfaceLevel const):
(WebCore::Document::compositeOperatorForBackgroundColor const):
(WebCore::Document::didAssociateFormControl):
(WebCore::Document::didAssociateFormControlsTimerFired):
(WebCore::Document::didLoadResourceSynchronously):
(WebCore::Document::ensurePlugInsInjectedScript):
(WebCore::Document::wrapCryptoKey):
(WebCore::Document::unwrapCryptoKey):
(WebCore::Document::hasFocus const):
(WebCore::Document::addPlaybackTargetPickerClient):
(WebCore::Document::removePlaybackTargetPickerClient):
(WebCore::Document::showPlaybackTargetPicker):
(WebCore::Document::playbackTargetPickerClientStateDidChange):
(WebCore::Document::shouldOpenExternalURLsPolicyToPropagate const):
(WebCore::Document::shouldEnforceHTTP09Sandbox const):
(WebCore::Document::shouldEnforceQuickLookSandbox const):
(WebCore::Document::applyQuickLookSandbox):
(WebCore::Document::shouldEnforceContentDispositionAttachmentSandbox const):
(WebCore::Document::scheduleRenderingUpdate):
(WebCore::Document::updateIntersectionObservations):
(WebCore::Document::gatherResizeObservations):
(WebCore::Document::deliverResizeObservations):
(WebCore::Document::setHasSkippedResizeObservations):
(WebCore::Document::setDir):
(WebCore::Document::fgColor const):
(WebCore::Document::alinkColor const):
(WebCore::Document::linkColorForBindings const):
(WebCore::Document::vlinkColor const):
(WebCore::Document::logger):
(WebCore::Document::updateAnimationsAndSendEvents):
(WebCore::Document::matchingAnimations):
(WebCore::Document::hideAllPopoversUntil):
(WebCore::Document::handlePopoverLightDismiss):
(WebCore::Document::didLogMessage):
(WebCore::Document::setServiceWorkerConnection):
(WebCore::Document::updateServiceWorkerClientData):
(WebCore::Document::navigateFromServiceWorker):
(WebCore::Document::frameWasDisconnectedFromOwner):
(WebCore::Document::hitTest):
(WebCore::Document::setAsRunningUserScripts):
(WebCore::Document::setPictureInPictureElement):
(WebCore::Document::notificationClient):
(WebCore::Document::graphicsClient):
(WebCore::Document::sessionID const):
(WebCore::Document::observeForContainIntrinsicSize):
(WebCore::Document::unobserveForContainIntrinsicSize):
(WebCore::Document::resetObservationSizeForContainIntrinsicSize):
(WebCore::Document::noiseInjectionPolicy const):
(WebCore::Document::noiseInjectionHashSalt const):
(WebCore::Document::mediaKeysStorageDirectory):
(WebCore::Document::protectedWindow const):
(WebCore::Document::protectedCachedResourceLoader const):
(WebCore::Document::protectedFontSelector const):
* Source/WebCore/dom/Document.h:
(WebCore::Document::setContextDocument):
(WebCore::Document::protectedParentDocument const):
(WebCore::Document::protectedTopDocument const):
* Source/WebCore/dom/Node.cpp:
(WebCore::Node::~Node):
* Source/WebCore/fileapi/BlobURL.cpp:
(WebCore::blobOwner):
* Source/WebCore/layout/layouttree/LayoutTreeBuilder.cpp:
(WebCore::Layout::printLayoutTreeForLiveDocuments):
* Source/WebCore/page/FrameDestructionObserver.h:
* Source/WebCore/page/FrameDestructionObserverInlines.h:
(WebCore::FrameDestructionObserver::protectedFrame const):
* Source/WebCore/page/LocalFrame.cpp:
(WebCore::Document::protectedView const):
* Source/WebCore/page/LocalFrameView.cpp:
(WebCore::LocalFrameView::protectedFrame const):
* Source/WebCore/page/LocalFrameView.h:
* Source/WebCore/page/MemoryRelease.cpp:
(WebCore::releaseNoncriticalMemory):
(WebCore::releaseCriticalMemory):
* Source/WebCore/page/mac/PageMac.mm:
(WebCore::Page::platformInitialize):
* Source/WebCore/rendering/HitTestResult.cpp:
(WebCore::HitTestResult::setScrollbar):
* Source/WebCore/rendering/HitTestResult.h:
* Source/WebCore/rendering/RenderObject.cpp:
(WebCore::printPaintOrderTreeForLiveDocuments):
(WebCore::printRenderTreeForLiveDocuments):
(WebCore::printLayerTreeForLiveDocuments):
(WebCore::printGraphicsLayerTreeForLiveDocuments):
* Source/WebCore/rendering/RenderView.h:
* Source/WebCore/testing/Internals.cpp:
(WebCore::Internals::allSVGSVGElements const):
(WebCore::Internals::numberOfLiveNodes const):
* Source/WebCore/workers/service/SWClientConnection.cpp:
(WebCore::SWClientConnection::postTaskForJob):
(WebCore::postMessageToContainer):
(WebCore::SWClientConnection::postMessageToServiceWorkerClient):
(WebCore::SWClientConnection::updateRegistrationState):
(WebCore::SWClientConnection::updateWorkerState):
(WebCore::SWClientConnection::fireUpdateFoundEvent):
(WebCore::SWClientConnection::setRegistrationLastUpdateTime):
(WebCore::SWClientConnection::setRegistrationUpdateViaCache):
(WebCore::SWClientConnection::updateBackgroundFetchRegistration):
(WebCore::updateController):
(WebCore::SWClientConnection::notifyClientsOfControllerChange):
(WebCore::SWClientConnection::clearPendingJobs):
(WebCore::SWClientConnection::registerServiceWorkerClients):
* Source/WebKit/WebProcess/InjectedBundle/InjectedBundle.cpp:
(WebKit::InjectedBundle::liveDocumentURLs):

Canonical link: https://commits.webkit.org/269557@main
